docs/troubleshooting.md

```diff
@@ -0,0 +1,255 @@
+
+```
+$ oc get opsrc -n openshift-marketplace
+
+NAME                  TYPE          ENDPOINT              REGISTRY              DISPLAYNAME           PUBLISHER   STATUS      MESSAGE                                       AGE
+certified-operators   appregistry   https://quay.io/cnr   certified-operators   Certified Operators   Red Hat     Succeeded   The object has been successfully reconciled   22h
+community-operators   appregistry   https://quay.io/cnr   community-operators   Community Operators   Red Hat     Succeeded   The object has been successfully reconciled   22h
+redhat-operators      appregistry   https://quay.io/cnr   redhat-operators      Red Hat Operators     Red Hat     Succeeded   The object has been successfully reconciled   22h
+```
+
+If the status of any of the OperatorSource is anything other than `Succeeded`, you can investigate the OperatorSource with 
+`oc describe opsrc <Name-of-OperatorSource> -n openshift-marketplace`
+
+If the OperatorSources look healthy, check that the CatalogSourceConfigs for the corresponding OperatorSources were created.
+
+```
+$ oc get csc -n openshift-marketplace
+
+NAME                  STATUS      MESSAGE                                       AGE
+certified-operators   Succeeded   The object has been successfully reconciled   23h
+community-operators   Succeeded   The object has been successfully reconciled   23h
+redhat-operators      Succeeded   The object has been successfully reconciled   23h
+```
+
+If the status of any of the CatalogSourceConfigs is anything other than `Succeeded`, you can inspect the CatalogSourceConfig with `oc describe csc <Name-of-CatalogSourceConfig>`
+
+If the CatalogSourceConfigs look healthy, make sure that there's a pod corresponding to each CatalogSourceConfig and that they are in the `Running` state.
+
+```
+$ oc get pods -n openshift-marketplace
+
+NAME                                    READY     STATUS    RESTARTS   AGE
+certified-operators-79f876c76d-mbtjg    1/1       Running   0          86m
+community-operators-854c564f5f-9vk2x    1/1       Running   0          86m
+marketplace-operator-6dcd66cb8c-2wmkz   1/1       Running   0          86m
+redhat-operators-647fd668fc-9ff5d       1/1       Running   0          86m
+
+```
+
+If any pod is in a state other than `Running` inspect the logs of the pod with `oc logs <Name-of-Pod>`.
+
+If all pods look healthy, make sure that there's a CatalogSource for each corresponding CatalogSourceConfigs.
+
+```
+$ oc get catalogsource -n openshift-marketplace 
+
+NAME                  NAME                  TYPE      PUBLISHER   AGE
+certified-operators   Certified Operators   grpc      Red Hat     23h
+community-operators   Community Operators   grpc      Red Hat     23h
+redhat-operators      Red Hat Operators     grpc      Red Hat     23h
+
+```
+
+// TODO 
+Inspect -o yaml for catalogsource 
+
+// TODO 
+oc logs catalog-operator | grep <Operator-name>
```

Comment:
The `catalog-operator` is only relevant when installing an operator, in the case that a subscription isn't being picked up (no status set, and no installPlan created)